### PR TITLE
Fixes #10: Support specific subnetworks

### LINKED ISSUE #10: Support specific subnetworks
The `google_compute_router_nat` resource supports specific subnetworks: it would be great to have this feature in the NAT module.

Example:
```hcl
resource "google_compute_router_nat" "main" {
  region     = "${google_compute_router.main.region}"	
  router     = "${google_compute_router.main.name}"
  nat_ip_allocate_option             = "AUTO_ONLY"
  source_subnetwork_ip_ranges_to_nat = "LIST_OF_SUBNETWORKS"
  subnetwork {
    name                    = "${data.google_compute_subnetwork.main.self_link}"
    source_ip_ranges_to_nat = ["ALL_IP_RANGES"]
  }
}
```

### PR DESCRIPTION
**Pull request by nick4fake**
Fixes #10 

Added subnetworks variable + additional integration tests covering it.

### CODE REVIEW

File: examples/advanced/main.tf [hunk 1]
```diff
@@ -15,16 +15,18 @@
...
+  region                             = var.region
+  router                             = var.router_name
+  name                               = "my-cloud-nat-${var.router_name}"
+  nat_ips                            = var.nat_addresses
+  min_ports_per_vm                   = "128"
+  icmp_idle_timeout_sec              = "15"
+  tcp_established_idle_timeout_sec   = "600"
+  tcp_transitory_idle_timeout_sec    = "15"
+  udp_idle_timeout_sec               = "15"
+  source_subnetwork_ip_ranges_to_nat = var.source_subnetwork_ip_ranges_to_nat
```
Comment (morgante) [inline examples/advanced/main.tf]: Examples should ideally require minimal external config.

We should directly incorporate the subnetwork setup into the example.

Comment (nick4fake) [reply examples/advanced/main.tf]: How do you see it?

**Comment by nick4fake** [comment]
image

**Comment by nick4fake** [comment]
image

**Comment by aaron-lane** [comment]
Fixed by #17

**Pull request closed without merging**